src/pyprediktormapclient/analytics_helper.py

```diff
@@ -117,7 +122,12 @@ def namespaces_as_list(self, list_of_dicts: List) -> List:
         return new_list
 
     @validate_call
-    def split_id(self, id: constr(pattern=r"^\d+:\d+:\S+$")):
+    def split_id(self, id: Any) -> dict:
+
+        pattern = r"^\d+:\d+:\S+$"
```

Comment:
If the pattern is the constant value, you can move it to the class body, so it's a class attribute that will be instantiated once, instead of being created each time you call the method.

```python
class AnalyticsHelper:
    ID_PATTERN = r"^\d+:\d+:\S+$"
    # ...
    @validate_call
    def split_id(self, id: Any) -> dict:
        if not re.match(self.ID_PATTERN, id):
        # ... the rest of your code
```